test/regression/int_pruf_plant_analysis.py

```diff
@@ -35,11 +48,200 @@ def test_monthly_inputs(self):
             reg_winddirection=True,
         )
         df = self.analysis._aggregate.df
+        df_rean = self.analysis._reanalysis_aggregate
 
         # Check the pre-processing functions
         self.check_process_revenue_meter_energy_monthly(df)
         self.check_process_loss_estimates_monthly(df)
-        self.check_process_reanalysis_data_monthly(df)
+        self.check_process_reanalysis_data_monthly(df, df_rean)
+
+    # Test reanalysis start and end dates depending on time resolution and end date argument
+    def test_reanalysis_aggregate(self):
```

Comment:
Because this test function is quite long, I might break into a monthly, daily, and hourly check to just to make a it a little bit easier to follow. I also didn't see a check where the start date is defined for ensuring we're getting what's expected, and that too late of starting date will fail, so I would highly recommend adding those checks in. Otherwise, this looks good.

Reply:
Good suggestion to split that test function into smaller functions, I agree it was pretty hard to follow. I only included an optional end date argument because the long-term correction always selects data starting X number of years before the last date in the reanalysis time series. So I figured everything could be defined using an end date argument and defining a start date wouldn't really accomplish anything. However, I did add another error checking step to make sure the raw reanalysis data contains enough time history for the LT correction, even if the end date argument is not defined. Does this cover the cases you thought we should check for?

Reply:
I like the split with it being monthly, daily, and hourly, thanks for making that change!